test/unit-distributed/test_communicator.cpp

```diff
@@ -531,8 +531,8 @@ TEST(communicator, ring)
     auto global_sources = g_context->distributed->gather_cell_labels_and_gids(local_sources);
 
     // construct the communicator
-    auto C = communicator(R, D, label_resolution_map(global_sources), label_resolution_map(local_targets), *g_context);
-
+    auto C = communicator(R, D, *g_context);
+    C.update_connections(R, D, label_resolution_map(global_sources), label_resolution_map(local_targets));
```

Comment:
this looks like a common pattern - would it make sense to add an overload à la
```
void communicator::update_connections(
    const connectivity& rec,
    const domain_decomposition& dom_dec,
    const cell_labels_and_gids&  local_sources);
```

Reply:
Actually, why not allow automatic conversion of sources to label map?

Reply:
yes, that too. but also default conversion from local to global?

Reply:
Hmm, upon surveying the situation, it's only ever used in the tests, and there only thrice.